src/main/java/io/github/thebusybiscuit/slimefun4/core/attributes/EnergyNetComponent.java

```diff
@@ -36,4 +43,104 @@
      */
     int getCapacity();
 
+    /**
+     * This returns whether this {@link EnergyNetComponent} can hold energy charges.
+     * It returns true if {@link #getCapacity()} returns a number greater than zero.
+     * 
+     * @return Whether this {@link EnergyNetComponent} can store energy.
+     */
+    default boolean isChargeable() {
+        return getCapacity() > 0;
+    }
+
+    /**
+     * This returns the currently stored charge at a given {@link Location}.
+     * 
+     * @param l
+     *            The target {@link Location}
+     * 
+     * @return The charge stored at that {@link Location}
+     */
+    default int getCharge(Location l) {
+        String charge = BlockStorage.getLocationInfo(l, "energy-charge");
+
+        if (charge != null) {
+            return Integer.parseInt(charge);
+        }
+        else {
+            return 0;
+        }
+    }
+
+    /**
+     * This method sets the charge which is stored at a given {@link Location}
+     * If this {@link EnergyNetComponent} is of type {@code EnergyNetComponentType.CAPACITOR}, then
+     * this method will automatically update the texture of this {@link Capacitor} as well.
+     * 
+     * @param l
+     *            The target {@link Location}
+     * @param charge
+     *            The new charge
+     */
+    default void setCharge(Location l, int charge) {
+        int capacity = getCapacity();
+
+        // This method only makes sense if we can actually store energy
+        if (capacity > 0) {
```

Comment:
isChargeable should be used here

Reply:
That would just be the same call to getCapacity() again. The point of this is to not get the capacity twice.

Reply:
Also, isChargeable() could be overridden. We don't know the internals of that method in that scenario and neither do we know the computational time of that operation.
For these methods we assume they are chargeable, otherwise that method shouldn't have been called in the first place, we don't need to know if it actually should be charged or not, this decision should have taken place in advance. We should only care if a valid number was given to work with because of maths...